handler/randomizer.go

```diff
@@ -0,0 +1,58 @@
+package handler
+
+import (
+	"crypto/sha1"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"math/big"
+)
+
+const (
+	ScheduleHourlyRandom   = "@hourly-random"
+	ScheduleDailyRandom    = "@daily-random"
+	ScheduleYearlyRandom   = "@yearly-random"
+	ScheduleAnnuallyRandom = "@annually-random"
+	ScheduleMonthlyRandom  = "@monthly-random"
+	ScheduleWeeklyRandom   = "@weekly-random"
+)
+
+// randomizeSchedule randomizes the given originalSchedule with a seed. The originalSchedule has to be one of the supported
+// '@x-random' predefined schedules, otherwise it returns an error with the original schedule unmodified.
+func randomizeSchedule(seed, originalSchedule string) (string, error) {
+	sumBase16 := calculateChecksumFromSeed(seed)
+
+	minute := new(big.Int).Set(sumBase16).Mod(sumBase16, big.NewInt(60))
+	hour := new(big.Int).Set(sumBase16).Mod(sumBase16, big.NewInt(24))
+	// A month can have between 27 and 31 days. Cron does not automatically schedule at the end of the month if 31 in February.
+	// To not cause a skip of a scheduled backup that could potentially raise alerts, we cap the day-of-month to 27 so it fits in all months.
+	dayOfMonth := new(big.Int).Set(sumBase16).Mod(sumBase16, big.NewInt(27))
+	month := new(big.Int).Set(sumBase16).Mod(sumBase16, big.NewInt(12))
+	weekday := new(big.Int).Set(sumBase16).Mod(sumBase16, big.NewInt(6))
+
+	effectiveSchedule := originalSchedule
+	switch originalSchedule {
+	case ScheduleHourlyRandom:
+		effectiveSchedule = fmt.Sprintf("%d * * * *", minute)
+	case ScheduleDailyRandom:
+		effectiveSchedule = fmt.Sprintf("%d %d * * *", minute, hour)
+	case ScheduleMonthlyRandom:
+		effectiveSchedule = fmt.Sprintf("%d %d %d * *", minute, hour, dayOfMonth.Add(dayOfMonth, big.NewInt(1)))
+	case ScheduleAnnuallyRandom:
+	case ScheduleYearlyRandom:
+		effectiveSchedule = fmt.Sprintf("%d %d %d %d *", minute, hour, dayOfMonth.Add(dayOfMonth, big.NewInt(1)), month.Add(month, big.NewInt(1)))
+	case ScheduleWeeklyRandom:
+		effectiveSchedule = fmt.Sprintf("%d %d * * %d", minute, hour, weekday)
+	default:
+		return originalSchedule, errors.New("unrecognized random schedule: " + originalSchedule)
```

Comment:
```suggestion
		return originalSchedule, fmt.Errorf("unrecognized random schedule: '%v'", originalSchedule)
```

I've learned in the past that it is good habit to enclose string values with ticks or quotes. It makes it easy to identify problems with misplaced whitespaces. I'm not certain if that even could apply here, but it doesn't hurt either IMO.